py-core/app/__init__.py

```diff
@@ -0,0 +1,247 @@
+
+    os.system(SCRIPT_FILE_PATH +
+              "run_conseil.sh " +
+              name +
+              " " +
+              str(data["rpc_port"]) +
+              " " +
+              str(data["network"]) +
+              " " +
+              str(data["conseil_port"]) +
+              " " +
+              str(data["postgres_port"])
+              )
+
+    os.system(SCRIPT_FILE_PATH +
+              "run_arronax.sh " +
+              name
+              )
+
+    # Store node process statistics
+    data["status"] = "running"
+
+    nodes.append(str(name))
+
+    add_node(data)
+
+    return redirect("/")
+
+
+@app.route("/stop_node", methods=["GET"])
+def stop_node():
+
+    os.system(SCRIPT_FILE_PATH +
+              "stop_node.sh " +
+              request.args.get('name')
+              )
+
+    os.system(SCRIPT_FILE_PATH +
+              "stop_conseil.sh " +
+              request.args.get('name')
+              )
+
+    os.system(SCRIPT_FILE_PATH +
+              "stop_arronax.sh " +
+              request.args.get('name')
+              )
+
+    os.system(SCRIPT_FILE_PATH +
+              "stop_postgres.sh " +
+              request.args.get('name')
+              )
+
+    update_status(request.args.get('name'), "stopped")
+    return render_template("node.html", node=get_node_data(request.args.get("name")))
+
+
+@app.route("/restart_node", methods=['GET'])
+def restart_node():
+    name = str(request.args.get("name"))
+    data = get_node_data(name)
+
+    os.system(SCRIPT_FILE_PATH +
+              "restart_node.sh " +
+              str(data["network"]) +
+              " " +
+              str(data["rpc_port"]) +
+              " " +
+              str(data["exposition_port"]) +
+              " " +
+              str(name) +
+              " " +
+              str(data["history_mode"])
+              )
+
+    os.system(SCRIPT_FILE_PATH +
+              "restart_conseil.sh " +
+              name +
+              " " +
+              str(data["rpc_port"]) +
+              " " +
+              str(data["network"]) +
+              " " +
+              str(data["conseil_port"])
+              )
+
+    os.system(SCRIPT_FILE_PATH +
+              "restart_arronax.sh " +
+              name
+              )
+
+    update_status(name, "running")
+    return redirect("/node?name=" + name)
+
+
+@app.route("/delete_node", methods=['GET'])
+def delete_node():
+    name = str(request.args.get("name"))
+    os.system(SCRIPT_FILE_PATH + "delete_node.sh " + name)
+    remove_node(name)
+    nodes.remove(name)
+    return redirect("/")
+
+
+@app.route("/node", methods=["GET"])
+def node_page():
+    return render_template("node.html", node=get_node_data(request.args.get("name")))
+
+
+@app.route("/rpc", methods=['GET'])
+def node_rpc_page():
+    return render_template("rpc.html", node=get_node_data(request.args.get("name")))
+
+
+if __name__ == "__main__":
+    setup_database()
+    nodes = list(map(lambda node: node[0], get_all_nodes()))
```

Comment:
This should be a function call executed every time the list of node names is required.